CRM/Odoosync/Upgrader.php

```diff
@@ -184,4 +221,85 @@ private function getSyncErrorTemplateWorkflowID() {
     return (int) $id;
   }
 
+  /**
+   * Enables all the option groups created by the extension
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function enableExtensionOptionGroups() {
+    foreach ($this->optionGroups as $optionGroupName) {
+      $this->toggleOptionGroupActivity($optionGroupName, TRUE);
+    }
+  }
+
+  /**
+   * Enables all the custom groups created by the extension
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function enableExtensionCustomGroups() {
+    foreach ($this->customGroups as $customGroupName) {
+      $this->toggleCustomGroupActivity($customGroupName, TRUE);
+    }
+  }
+
+  /**
+   * Disables all the custom groups created by the extension
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function disableExtensionCustomGroups() {
+    foreach ($this->customGroups as $customGroupName) {
+      $this->toggleCustomGroupActivity($customGroupName, FALSE);
+    }
+  }
+
+  /**
+   * Disables all the option groups created by the extension
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function disableExtensionOptionGroups() {
+    foreach ($this->optionGroups as $optionGroupName) {
+      $this->toggleOptionGroupActivity($optionGroupName, FALSE);
+    }
+  }
+
+  /**
+   * Sets is_active for OptionGroups
+   *
+   * @param string $name
+   * @param bool $isActive
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function toggleOptionGroupActivity($name, $isActive) {
+    civicrm_api3('OptionGroup', 'get', [
+      'name' => $name,
+      'api.OptionGroup.create' => [
+        'id' => '$value.id',
+        'is_active' => (int) $isActive
+      ],
+    ]);
+  }
+
+  /**
+   * Sets is_active for CustomGroups
+   *
+   * @param string $name
+   * @param bool $isActive
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function toggleCustomGroupActivity($name, $isActive) {
+    $customGroup = civicrm_api3('CustomGroup', 'getsingle', [
+      'name' => $name,
+      'return' => ['id']
+    ]);
+
+    if (isset($customGroup['id'])) {
+      CRM_Core_BAO_CustomGroup::setIsActive((int) $customGroup['id'], $isActive);
```

Comment:
why not to use API chaining as u did in  toggleOptionGroupActivity , is it does not work for custom groups ?

Reply:
If we use API chaining for toggle Custom Group it returns an Error on extension install and a Notice on extension enable/disable. But API chaining for toggle Option Group works good.
Therefore I've decided to write as it is now.